test/specs/EventsSpec.js

```diff
@@ -21,325 +21,355 @@ describe("Event Tests", function() {
     });
 
     describe("Mouse Events", function() {
+      var spy;
 
-      it("'slideStart' event is triggered properly and can be binded to", function() {
-        testSlider.on('slideStart', function() {
-          flag = true;
-        });
-        testSlider.data('slider')._mousedown(mouse);
-        expect(flag).toBeTruthy();
+      beforeEach(function() {
+        spy = jasmine.createSpy('spy');
       });
 
-      it("'slide' event is triggered properly and can be binded to", function() {
-        testSlider.on('slide', function() {
-          flag = true;
+      it("'slideStart' event is triggered properly and can be binded to", function(done) {
+        testSlider.on('slideStart', spy);
+        testSlider.data('slider')._mousedown(mouse);
+        setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
         });
+      });
+
+      it("'slide' event is triggered properly and can be binded to", function(done) {
+        testSlider.on('slide', spy);
         testSlider.data('slider')._mousemove(mouse);
-        expect(flag).toBeTruthy();
+        setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
+        });
       });
 
-      it("'slide' event sets the right value on the input", function() {
+      it("'slide' event sets the right value on the input", function(done) {
         testSlider.on('slide', function() {
-          flag = true;
           expect(isNaN(testSlider.val())).not.toBeTruthy();
+          done();
         });
         testSlider.data('slider')._mousemove(mouse);
-        expect(flag).toBeTruthy();
       });
 
-      it("'slide' event value and input value properties are synchronous", function() {
+      it("'slide' event value and input value properties are synchronous", function(done) {
         testSlider.on('slide', function(e) {
-          flag = true;
           expect(e.value.toString()).toEqual(this.value);
+          done();
         });
         testSlider.slider("setValue", 3, true, false);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'change' event value and input value properties are synchronous", function() {
+      it("'change' event value and input value properties are synchronous", function(done) {
         testSlider.on('change', function(e) {
-          flag = true;
           expect(e.value.newValue.toString()).toEqual(testSlider.val());
```

Comment:
This test is testing whether the `change` event is triggered synchronously. The `done` callback should be removed.